models/issue_tracked_time.go

```
@@ -0,0 +1,118 @@
+	}
+	if opts.RepositoryID != 0 {
+		cond = cond.And(builder.Eq{"issue.repo_id": opts.RepositoryID})
+	}
+	return cond
+}
+
+// GetTrackedTimes returns all tracked times that fit to the given options.
+func GetTrackedTimes(options FindTrackedTimesOptions) (trackedTimes []*TrackedTime, err error) {
+	if options.RepositoryID > 0 {
+		err = x.Join("INNER", "issue", "issue.id = tracked_time.issue_id").Where(options.ToCond()).Find(&trackedTimes)
+		return
+	}
+	err = x.Where(options.ToCond()).Find(&trackedTimes)
+	return
+}
+
+// BeforeInsert will be invoked by XORM before inserting a record
+// representing this object.
+func (t *TrackedTime) BeforeInsert() {
+	t.CreatedUnix = time.Now().Unix()
+}
+
+// AddTime will add the given time (in seconds) to the issue
+func AddTime(userID int64, issueID int64, time int64) (*TrackedTime, error) {
+	tt := &TrackedTime{
+		IssueID: issueID,
+		UserID:  userID,
+		Time:    time,
+	}
+	if _, err := x.Insert(tt); err != nil {
+		return nil, err
+	}
+	comment := &Comment{
+		IssueID:  issueID,
+		PosterID: userID,
+		Type:     CommentTypeAddTimeManual,
+		Content:  secToTime(time),
+	}
+	if _, err := x.Insert(comment); err != nil {
+		return nil, err
+	}
+	return tt, nil
+}
+
+// TotalTimes returns the spent time for each user by an issue
+func TotalTimes(options FindTrackedTimesOptions) (map[*User]string, error) {
+	trackedTimes, err := GetTrackedTimes(options)
+	if err != nil {
+		return nil, err
+	}
+	//Adding total time per user ID
+	totalTimesByUser := make(map[int64]int64)
+	for _, t := range trackedTimes {
+		if total, ok := totalTimesByUser[t.UserID]; !ok {
+			totalTimesByUser[t.UserID] = t.Time
+		} else {
+			totalTimesByUser[t.UserID] = total + t.Time
+		}
+	}
+
+	totalTimes := make(map[*User]string)
+	//Fetching User and making time human readable
+	for userID, total := range totalTimesByUser {
+		user, err := GetUserByID(userID)
+		if err != nil || user == nil {
+			continue
```

Please don't silently ignore errors. We should check if `err` is `ErrUserNotExist`, and if not return `nil, err`.

Also, I believe the `user == nil` check is unnecessary; `GetUserByID(..)` should never return `nil, nil`.